include/boost/ut.hpp

```diff
@@ -1025,7 +1025,12 @@ struct eq_ : op {
 
           if constexpr (type_traits::has_static_member_object_value_v<TLhs> and
                         type_traits::has_static_member_object_value_v<TRhs>) {
+#if defined(_MSC_VER) && !defined(__clang__)
+            // for some reason, accessing the static member via :: does not compile on MSVC
```

Comment:
instead #if/#else why not simply switch to lhs.value @ rhs.value?

Reply:
That was my first attempt, but then the less-than operator does not compile on clang anymore. It interprets the `<` in `lhs.value < rhs.value` as opening brace for a template. For some reason, this code seems to be really hard to parse for compilers. We could probably use the `#if/else#` only for the less-than operator and simply switch to `lhs.value op rhs.value` for the other operators.